core/src/main/scala/org/genivi/sota/core/db/UpdateSpecs.scala

```diff
@@ -171,13 +158,18 @@ object UpdateSpecs {
   }
 
   /**
-    * Set status of a given update to 'In-Flight'
-    * @param vin the vin associated with the desired update
-    * @param uuid the uuid of the update whose status should be changed
+    * Rewrite in the DB the status of an [[UpdateSpec]], ie for a ([[UpdateRequest]], VIN) combination.
+    */
+  def setStatus(spec: UpdateSpec, newStatus: UpdateStatus) : DBIO[Int] = {
+    setStatus(spec.vin, spec.request.id, newStatus)
+  }
+
+  /**
+    * Rewrite in the DB the status of an [[UpdateSpec]], ie for a ([[UpdateRequest]], VIN) combination.
     */
-  def setStatus(vin: Vehicle.Vin, uuid: Refined[String, Uuid], newStatus: UpdateStatus): DBIO[Int] = {
+  def setStatus(vin: Vehicle.Vin, updateRequestId: UUID, newStatus: UpdateStatus): DBIO[Int] = {
     (for {
-      r <- updateSpecs.filter(us => us.vin === vin && us.requestId === uuid)
+      r <- updateSpecs.filter(us => us.vin === vin && us.requestId === updateRequestId)
     } yield r.status).update(newStatus).transactionally
```

Comment:
As long as we are here, I think this could be written clearer with:

```scala
    updateSpecs
      .filter(us => us.vin === vin && us.requestId === updateRequestId)
      .map(_.status)
      .update(newStatus)
      .transactionally
```